mlflow/tracking/client.py

```diff
@@ -2681,14 +2682,19 @@ def get_model_version_stages(
         """
         return ALL_STAGES
 
-    def set_model_version_tag(self, name: str, version: str, key: str, value: Any) -> None:
+    def set_model_version_tag(
+        self, name: str, version: str = None, key: str = None, value: Any = None, stage: str = None
+    ) -> None:
         """
         Set a tag for the model version.
+        When stage is set, tag will be set for latest model version of the stage.
+        Setting both version and stage parameter will result in error.
 
         :param name: Registered model name.
         :param version: Registered model version.
-        :param key: Tag key to log.
-        :param value: Tag value to log.
+        :param key: Tag key to log. key is required.
+        :param value: Tag value to log. value is required.
```

Comment:
Maybe it's just me but this comment is a bit

Reply:
@harupy @dbczumar func definition does not align with use case requirement. Even though func def allows `key` and `value` to be optional/None, operation will not be successful if `key` or `value` is optional/None. Let me know if you have better doc string phrase or change implementation.

Reply:
I think this is the best we can do for now. Thanks @subramaniam02 ! :)